### PR TITLE
Simplify protocol removals

### PR DESCRIPTION
**Pull request by jecisc**
This commit simplifies the removal of protocols. Here is the list of changes:

- #removeProtocol: was removing only empty protocols while we have #removeProtocolIfEmpty:. Thus I deprecated it because I prefer to have a selector explicitly saying it will work only if the protocol is empty.
- #removeProtocolIfEmpty: now accept a real protocol as parameter. This will help with the reified Protocol integration 
- #removeProtocolNamed: became obsolete with the previous improvement and is now deprecated
- #removeEmptyProtocols has been simplified to just iterate over the protocols and call #removeProtocolIfEmpty:. Also there was a bug where protocol removals were announced twice and this does not happen anymore.

The API of ClassOrganization lost 2 new methods and this will help with its future inlining :)

### CODE REVIEW

File: src/Kernel/ClassOrganization.class.st [hunk 1]
```diff
@@ -91,7 +91,7 @@ ClassOrganization >> classify: selector under: protocol [
 	oldProtocols := self protocolsOfSelector: selector.
 
 	self classify: selector inProtocolNamed: protocolName.
-	(oldProtocols select: #canBeRemoved) do: [ :e | self removeProtocol: e ].
+	(oldProtocols select: #canBeRemoved) do: [ :e | self removeProtocolIfEmpty: e ].
```
Comment (privat) [inline src/Kernel/ClassOrganization.class.st]: `#canBeRemoved` → `#isEmpty` ? and if `removeProtocolIfEmpty` is worth its name, maybe the select is superfluous.

File: src/Kernel/ClassOrganization.class.st [hunk 2]
```diff
@@ -305,7 +305,7 @@ ClassOrganization >> protocolsOfSelector: aSelector [
 	^ (self protocols select: [:each | each includesSelector: aSelector ]) asArray
 ]
 
-{ #category : #'backward compatibility' }
+{ #category : #removing }
```
Comment (privat) [inline src/Kernel/ClassOrganization.class.st:310]: why protocol change?

**Review by privat — APPROVED** [review]

File: src/Kernel/ClassOrganization.class.st (hunk 1 above)
Comment (jecisc) [reply src/Kernel/ClassOrganization.class.st]: Done thank you for the catch

File: src/Kernel/ClassOrganization.class.st (hunk 2 above)
Comment (jecisc) [reply src/Kernel/ClassOrganization.class.st:310]: I moved it because it is used by a lot of things so I don't think it is just for backward compatibility

**Review by MarcusDenker — APPROVED** [review]

**Pull request merged**